tests/ApiResponse.spec.ts

```diff
@@ -0,0 +1,45 @@
+import { AxiosResponse } from 'axios';
+import { ApiResponse } from '../src/common/ApiResponse';
+
+test('ApiResponse has the correct pageCount value', () => {
+    const baseUrl = 'https://test.com/path';
+    const expectedPage = 7;
+
+    const axiosResponse = {
+        data: null,
+        status: 200,
+        headers: {
+            link: `<${baseUrl}?page=1>; rel="first", <${baseUrl}?page=${expectedPage}>; rel="last"`,
+        },
+    } as AxiosResponse<null>;
+
+    const response = new ApiResponse<null>(axiosResponse);
+
+    expect(response.pageCount).toBe(expectedPage);
+});
+
+test('ApiResponse do not fail if no have Link header', () => {
+    const axiosResponse = {
```

Comment:
`if doesn't have`
`if Link header not present`
`if no Link header is found`
`if no Link header present`
...